bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/utils/Converter.java

```diff
@@ -12,89 +12,77 @@
-import com.google.gson.JsonSyntaxException;
-import com.google.gson.reflect.TypeToken;
-
 /**
  * The {@link Converter} Conversion Helpers
  *
  * @author Bernd Weymann - Initial contribution
+ * @author Martin Grassl - extract some methods to other classes
  */
 @NonNullByDefault
-public class Converter {
-    public static final Logger LOGGER = LoggerFactory.getLogger(Converter.class);
+public interface Converter {
+    static final Logger LOGGER = LoggerFactory.getLogger(Converter.class);
 
-    public static final String DATE_INPUT_PATTERN_STRING = "yyyy-MM-dd'T'HH:mm:ss";
-    public static final DateTimeFormatter DATE_INPUT_PATTERN = DateTimeFormatter.ofPattern(DATE_INPUT_PATTERN_STRING);
-    public static final DateTimeFormatter LOCALE_ENGLISH_TIMEFORMATTER = DateTimeFormatter.ofPattern("hh:mm a",
+    static final String DATE_INPUT_PATTERN_STRING = "yyyy-MM-dd'T'HH:mm:ss";
+    static final DateTimeFormatter DATE_INPUT_PATTERN = DateTimeFormatter.ofPattern(DATE_INPUT_PATTERN_STRING);
+    static final DateTimeFormatter LOCALE_ENGLISH_TIMEFORMATTER = DateTimeFormatter.ofPattern("hh:mm a",
             Locale.ENGLISH);
-    public static final SimpleDateFormat ISO_FORMATTER = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS");
-
-    private static final Gson GSON = new Gson();
-    private static final Vehicle INVALID_VEHICLE = new Vehicle();
-    private static final String SPLIT_HYPHEN = "-";
-    private static final String SPLIT_BRACKET = "\\(";
-    private static final String VIN_PATTERN = "\"vin\":";
-    private static final String VEHICLE_LOCATION_PATTERN = "\"vehicleLocation\":";
-    private static final String VEHICLE_LOCATION_REPLACEMENT = "\"vehicleLocation\": {\"coordinates\": {\"latitude\": 1.1,\"longitude\": 2.2},\"address\": {\"formatted\": \"anonymous\"},\"heading\": -1}";
-    private static final char OPEN_BRACKET = "{".charAt(0);
-    private static final char CLOSING_BRACKET = "}".charAt(0);
+    static final SimpleDateFormat ISO_FORMATTER = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS");
 
-    // https://www.baeldung.com/gson-list
-    public static final Type VEHICLE_LIST_TYPE = new TypeToken<ArrayList<Vehicle>>() {
-    }.getType();
-    public static int offsetMinutes = -1;
+    static final String SPLIT_HYPHEN = "-";
+    static final String SPLIT_BRACKET = "\\(";
 
-    public static String zonedToLocalDateTime(String input) {
-        try {
-            ZonedDateTime d = ZonedDateTime.parse(input).withZoneSameInstant(ZoneId.systemDefault());
-            return d.toLocalDateTime().format(Converter.DATE_INPUT_PATTERN);
-        } catch (Exception e) {
-            LOGGER.debug("Unable to parse date {} - {}", input, e.getMessage());
+    static State zonedToLocalDateTime(@Nullable String input) {
+        if (input != null && !input.isEmpty()) {
+            try {
+                String dateString = ZonedDateTime.parse(input).withZoneSameInstant(ZoneId.systemDefault())
```

Comment:
yes, might be better. The timestamps provided in the JSON returned by the BMW API are in UTC. I just took over the code from the previous developer. How can I get OpenHab Timezone?

Reply:
With dependency injection: https://www.openhab.org/javadoc/latest/org/openhab/core/i18n/timezoneprovider

Reply:
You can obtain if from [TimeZoneProvider](https://www.openhab.org/javadoc/latest/org/openhab/core/i18n/timezoneprovider).

See for example:
https://github.com/openhab/openhab-addons/blob/597f01efe155ce3573205b3e79c8133d7906e339/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/factory/EnergiDataServiceHandlerFactory.java#L50-L56

Reply:
thx, trying. And I combine the change with the recommendation to use a pre-defined pattern for the date format. It turns out it is not so easy, because with every format it throws another exception when parsing. But I'll try further tomorrow.

Reply:
done, looks like it is working nicely!